extensions/amp-story/1.0/test/test-amp-story.js

```diff
@@ -1198,6 +1199,147 @@ describes.realWin(
         });
       });
 
+      describe('amp-story-subscriptions navigation', () => {
+        let subscriptionsEl;
+
+        beforeEach(async () => {
+          toggleExperiment(win, 'amp-story-subscriptions', true);
+
+          await createStoryWithPages(4, [
+            'cover',
+            'page-1',
+            'page-2',
+            'page-3',
+          ]);
+          subscriptionsEl = win.document.createElement(
+            'amp-story-subscriptions'
+          );
+          story.element.appendChild(subscriptionsEl);
+
+          // buildCallback() could be omitted as long as createStoryWithPages()
+          // is called because the getImpl() call inside it.
+          await story.layoutCallback();
+        });
+
+        it('should continue navigating to locked pages after the subscription state gets resolved to granted', async () => {
+          // Block access with pending subscriptions state.
+          await story.switchTo_('page-1');
```

Comment:
lets dispatch an event here instead of calling a private method

Reply:
Done.